### PR TITLE
improve trust handling in tasks

### PR DESCRIPTION
**Pull request by meganrogge**
fix https://github.com/microsoft/vscode/issues/160013
fix https://github.com/microsoft/vscode/issues/160368
fix https://github.com/microsoft/vscode/issues/156183

### CODE REVIEW

File: src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts [hunk 1]
```diff
@@ -34,23 +35,33 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 		@IOpenerService private readonly _openerService: IOpenerService,
 		@INotificationService private readonly _notificationService: INotificationService) {
 		super();
-		this._tryRunTasks();
+		if (this._workspaceTrustManagementService.isWorkspaceTrusted()) {
+			this._tryRunTasks();
+		}
+		this._register(this._workspaceTrustManagementService.onDidChangeTrust(async trusted => {
```
Comment (alexr00) [inline src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts:41]: Do we still need to listen to this event if the above `isWorkspaceTrusted` check is true?

**Review by alexr00 — APPROVED** [review]

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 2]
```diff
@@ -1226,10 +1225,6 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			} else {
 				executeTaskResult = await this._executeTask(task, resolver, runSource);
 			}
-			if (runSource === TaskRunSource.User) {
-				const workspaceTasks = await this.getWorkspaceTasks();
-				RunAutomaticTasks.runWithPermission(this, this._storageService, this._notificationService, this._workspaceTrustManagementService, this._openerService, this._configurationService, workspaceTasks);
```
[the remark this answers is not in the document]
Comment (meganrogge) [reply src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:1231]: Yes that does make sense. Here is the model I think also makes sense now - lmk if you prefer the above - my logic behind it is if I were to have automatic tasks in an untrusted workpsace, i'd expect them to run when the workspace becomes trusted.

Screen Shot 2022-09-08 at 9 11 03 AM (2)

File: src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts:41]: good pt 👍🏼

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexr00) [reply src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:1231]: > my logic behind it is if I were to have automatic tasks in an untrusted workpsace, i'd expect them to run when the workspace becomes trusted.

That make sense to me. I think it's also nice to ask the user if they want to run automatic tasks around the time that they've just used tasks too, but it's up to you.

**Pull request merged**